### PR TITLE
Uniques: Allowing multiple approvals

### PR DESCRIPTION
**Pull request by Szegoo**
Solves the `new approvals structure` task from #11783

In short, this PR allows the item owner to make approvals to multiple accounts.

Note: I took a bit of a different approach than proposed in #11791. I believe that this is a bit simpler than storing a `BoundedBTreeMap` for approvals.

### CODE REVIEW

**Comment by jsidorenko** [comment]
Hi @Szegoo , thanks for your contribution!
I'm thinking it's better to keep the existing Uniques pallet as feature-complete and start pushing all the new changes into V2. 
I've created a new branch for this [js/uniques-v2-main-branch](https://github.com/paritytech/substrate/tree/js/uniques-v2-main-branch), this is a new pallet for all V2 works.
It would be good if you could apply your changes on top of that pallet and point your PR to my branch.

**Comment by jsidorenko** [comment]
The reason I wanted to use `BoundedBTreeMap` is because of the new structure of approvals I'm proposing: I think each approval should have an optional deadline (the block number) set.
Pls check my prev implementation for reference:
https://github.com/paritytech/substrate/commit/707d69bca379b53ee560a803e00ce181acc0de55
https://github.com/paritytech/substrate/commit/a705c225223ccf3d3d1715ecaf21b054c33a6a10

**Comment by Szegoo** [comment]
@jsidorenko What would be the benefit of creating a `BoundedBTreeMap`, cause we could also probably add deadline like this: 

```rust
/// The approved transferrer of this item, if one is set.
pub(super) approved:
	BoundedVec<(AccountId, Option<<T as frame_system::Config>::BlockNumber>), ApprovalsLimit>,
```

I am going to use `BoundedBTreeMap` if you still think we should use that, I am just interested if there is a reason behind this decision.

**Comment by jsidorenko** [comment]
`BoundedBTreeMap` simplifies the management of approvals:
1) when you need to find an approval, you will simply call `item.approvals.get(&sender)` instead of `item.approvals.get_key_value(&sender)`
2) once you insert, you don't need to check if approval for the same account exists
3) when you remove, you will simply call `item.approvals.remove(&delegate)` instead of `if let Some(pos) = item.approvals.iter().position(|ap| delegate == ap.who) { item.approvals.remove(pos); }`

File: frame/uniques/src/lib.rs [hunk 1]
```diff
@@ -321,16 +355,25 @@ pub mod pallet {
...
 		/// New metadata has been set for a `collection`.
 		CollectionMetadataSet {
 			collection: T::CollectionId,
 			data: BoundedVec<u8, T::StringLimit>,
 			is_frozen: bool,
 		},
 		/// Metadata has been cleared for a `collection`.
-		CollectionMetadataCleared { collection: T::CollectionId },
+		CollectionMetadataCleared {
+			collection: T::CollectionId,
```
Comment (jsidorenko) [inline frame/uniques/src/lib.rs:375]: Try to run `rustup update` and then: `cargo +nightly fmt`
The latest fmt shouldn't change the syntax that way

**Comment by Szegoo** [comment]
@jsidorenko I opened a new PR with these changes to the [js/uniques-v2-main-branch](https://github.com/paritytech/substrate/tree/js/uniques-v2-main-branch), Should I close this?

**Pull request closed without merging**